frontend/src/components/widgets/CameraImageInput/CameraImageInput.tsx

```diff
@@ -0,0 +1,499 @@
+    this.removeCapture = this.removeCapture.bind(this)
+    this.onMediaError = this.onMediaError.bind(this)
+    this.onUserMedia = this.onUserMedia.bind(this)
+  }
+
+  public capture(): void {
+    const imageSrc = this.webcamRef.current.getScreenshot()
+    this.setState({
+      imgSrc: imageSrc,
+    })
+
+    urltoFile(imageSrc, `${Date.now().toString()}.jpg`)
+      .then(file => this.uploadFile(file))
+      .catch(err => {
+        console.log(err)
+      })
+  }
+
+  public removeCapture(): void {
+    if (this.state.files.length === 0) {
+      return
+    }
+
+    const fileId = this.state.files[this.state.files.length - 1].id
+    this.deleteFile(fileId)
+
+    this.setState({
+      imgSrc: null,
+    })
+  }
+
+  public onMediaError(): void {
+    this.setState({
+      webcamRequestState: "error",
+    })
+  }
+
+  public onUserMedia(): void {
+    this.setState({
+      webcamRequestState: "success",
+    })
+  }
+
+  get initialValue(): State {
+    const emptyState = {
+      files: [],
+      newestServerFileId: 0,
+      imgSrc: null,
+      webcamRequestState: "pending",
+    }
+    const { widgetMgr, element } = this.props
+
+    const widgetValue = widgetMgr.getFileUploaderStateValue(element)
+    if (widgetValue == null) {
+      return emptyState
+    }
+
+    const { maxFileId, uploadedFileInfo } = widgetValue
+    if (maxFileId == null || maxFileId === 0 || uploadedFileInfo == null) {
+      return emptyState
+    }
+
+    return {
+      imgSrc: null,
+      webcamRequestState: "pending",
+      files: uploadedFileInfo.map(f => {
+        const name = f.name as string
+        const size = f.size as number
+        const serverFileId = f.id as number
+
+        return new UploadFileInfo(name, size, this.nextLocalFileId(), {
+          type: "uploaded",
+          serverFileId,
+        })
+      }),
+      newestServerFileId: Number(maxFileId),
+    }
+  }
+
+  public componentWillUnmount(): void {
+    this.formClearHelper.disconnect()
+  }
+
+  /**
+   * Return the FileUploader's current status, which is derived from
+   * its state.
+   */
+  public get status(): FileUploaderStatus {
+    const isFileUpdating = (file: UploadFileInfo): boolean =>
+      file.status.type === "uploading"
+
+    // If any of our files is Uploading or Deleting, then we're currently
+    // updating.
+    if (this.state.files.some(isFileUpdating)) {
+      return "updating"
+    }
+
+    return "ready"
+  }
+
+  public componentDidUpdate = (prevProps: Props): void => {
+    const { element, widgetMgr } = this.props
+
+    // Widgets are disabled if the app is not connected anymore.
+    // If the app disconnects from the server, a new session is created and users
+    // will lose access to the files they uploaded in their previous session.
+    // If we are reconnecting, reset the file uploader so that the widget is
+    // in sync with the new session.
+    if (prevProps.disabled !== this.props.disabled && this.props.disabled) {
```

Comment:
It looks like we are using this.props.disabled when it's always set to false? How does this work?